### PR TITLE
New book#18

### PR DESCRIPTION
**Pull request by mgoodhart5**
Driver navigator'd user story 12.
Adds new book page and form.
Checks book title and author names for title case.
Checks book titles and author names for uniqueness in the system.
Checks ability to add multiple authors.
All tests passing with 100% coverage.

@iandouglas @dionew1
could you guys please take a look at this and make sure we are on the right path?
in particular wondering about the books controller. Thanks!!

### CODE REVIEW

File: app/controllers/books_controller.rb [hunk 1]
```diff
@@ -2,8 +2,46 @@ class BooksController < ApplicationController
...
+      new_author = Author.new(name: author)
+      if new_author.save
+        multiple_authors << new_author
+      else
+        multiple_authors << Author.find_by(name: author_params[:name])
+      end
+    end
+    book = Book.new(book_params)
+    if book.save
+      multiple_authors.each do |author|
```
Comment (iandouglas) [inline app/controllers/books_controller.rb]: I think you could shorten this while block of code as
book.authors = multiple_authors

File: app/controllers/books_controller.rb [hunk 2]
```diff
@@ -2,8 +2,46 @@ class BooksController < ApplicationController
...
+
+  def create
+    whatever = author_params[:name].split(", ")
+    multiple_authors = []
+    whatever.each do |author|
+      new_author = Author.new(name: author)
+      if new_author.save
+        multiple_authors << new_author
+      else
+        multiple_authors << Author.find_by(name: author_params[:name])
```
Comment (iandouglas) [inline app/controllers/books_controller.rb]: You might want to do this lookup before trying to make/save a new author, unless you have a uniqueness validation on the author name.

**Comment by asmolentzov** [comment]
closes #44 
closes #18

**Comment by asmolentzov** [comment]
closes #18

**Pull request merged**